src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1156,19 +1156,37 @@ export class ResultsViewPageStore {
 
     readonly samples = remoteData({
         await: () => [
-            this.studyToSampleIds
+            this.studyToDataQueryFilter
         ],
-        invoke: () => {
+        invoke: async() => {
             let sampleIdentifiers: SampleIdentifier[] = [];
-            _.each(this.studyToSampleIds.result, (sampleIds: { [sampleId: string]: boolean }, studyId: string) => {
-                sampleIdentifiers = sampleIdentifiers.concat(Object.keys(sampleIds).map(sampleId => ({
-                    sampleId,
-                    studyId
-                })));
-            });
-            return client.fetchSamplesUsingPOST({
-                sampleIdentifiers
+            let sampleListIds: string[] = [];
+            _.each(this.studyToDataQueryFilter.result, (dataQueryFilter: IDataQueryFilter, studyId: string) => {
+                if (dataQueryFilter.sampleIds) {
+                    sampleIdentifiers = sampleIdentifiers.concat(dataQueryFilter.sampleIds.map(sampleId => ({
+                        sampleId,
+                        studyId
+                    })));
+                } else if (dataQueryFilter.sampleListId) {
+                    sampleListIds.push(dataQueryFilter.sampleListId);
+                }
             });
+            let promises:Promise<Sample[]>[] = [];
+            if (sampleIdentifiers.length) {
```

Comment:
i see, so there is possiblity we have BOTH sample lists and sampleIdentifiers?

Reply:
I think the way it currently is, there is no such possibility - but it's easy to imagine this would happen down the line if our cohort creation UI becomes more sophisticated.

Reply:
but is it possible now we would somehow have overlapping?  i.e. both list and samples contained in that list?

Reply:
no thats not possible